tasks/orca_catalog_reporting/orca_catalog_reporting.py

```diff
@@ -125,19 +118,19 @@ def get_catalog_sql() -> text:
                 granules.id,
                 granules.collection_id, 
                 granules.cumulus_granule_id, 
-                granules.cumulus_create_time, 
+                EXTRACT(EPOCH FROM granules.cumulus_create_time AT TIME ZONE 'UTC') * 1000 as cumulus_create_time, 
                 granules.execution_id, 
-                granules.ingest_time, 
-                granules.last_update
+                EXTRACT(EPOCH FROM granules.ingest_time AT TIME ZONE 'UTC') * 1000 as ingest_time, 
+                EXTRACT(EPOCH FROM granules.last_update AT TIME ZONE 'UTC') * 1000 as last_update
             FROM
             (SELECT DISTINCT
                 granules.cumulus_granule_id
             FROM granules
             WHERE 
                 (:granule_id is null or cumulus_granule_id=ANY(:granule_id)) and 
                 (:collection_id is null or collection_id=ANY(:collection_id)) and 
-                (:start_timestamp is null or cumulus_create_time>=:start_timestamp) and 
-                (:end_timestamp is null or cumulus_create_time<:end_timestamp)
+                (:start_timestamp is null or EXTRACT(EPOCH FROM cumulus_create_time AT TIME ZONE 'UTC') * 1000>=:start_timestamp) and 
```

Comment:
After going through the remaining of the code and seeing you want the output to be an integer in milliseconds, I recommend changing the EPOCH lines to `(EXTRACT(EPOCH FROM date_trunc('milliseconds', cumulus_create_time) AT TIME ZONE 'UTC')*1000)::bigint` This will prevent any rounding errors and convert the value to an integer. This will also mean you will not need to do conversions in python since the driver will handle it for you.

Reply:
Switched.